packages/user/src/models/invite.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  BeforeCreate,
+  Entity,
+  OnInit,
+  PrimaryKey,
+  Property,
+  Index,
+  Filter,
+  OptionalProps,
+} from "@mikro-orm/core"
+
+import { DALUtils, generateEntityId } from "@medusajs/utils"
+import { DAL } from "@medusajs/types"
+
+type OptionalFields =
+  | "metadata"
+  | "accepted"
+  | DAL.SoftDeletableEntityDateColumns
+@Entity({ tableName: "invite" })
+@Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
+export default class Invite {
+  [OptionalProps]: OptionalFields
+
+  @PrimaryKey({ columnType: "text" })
+  id: string
+
+  @Index({
+    name: "invite_user_identifier_index",
+    expression: `create unique index "invite_user_identifier_index" on "invite" ("user_identifier") where deleted_at is null;`,
+  })
+  @Property({ columnType: "text" })
+  user_identifier: string
```

Comment:
think it is reasonable to just stick with `email`. Flow would be something like this:

1. User invites new user - enters their email.
2. New user receives invite link on email with a token.
3. If user wants to authenticate with e.g. Google they complete sign in flow.
4. Once authenticated with Google they pass the invite token as part of the user creation call.
5. AuthUser is associated with the new User. 

I.e., the email is not used to tie a the AuthUser's entity id, but simply ensures there is a trusted destination the invite token can be sent to.

Reply:
Email-password could also be facilitated through the user creation request with an additional step in the workflow to create the auth user.